code/modules/uplink/uplink_items.dm

```diff
@@ -80,15 +81,31 @@
 	/// Weight of the category. Used to determine the positioning in the uplink. High weight = appears first
 	var/weight = 0
 
+/// Returns by how much percentage do we reduce the price of the selected item
 /datum/uplink_item/proc/get_discount()
-	return pick(4;0.75,2;0.5,1;0.25)
 
+	var/selected_discount = pick_weight(list(
+		"small_discount" = 4,
+		"average_discount" = 2,
+		"big_discount" = 1,
+	))
+
+	switch(selected_discount)
```

Comment:
```suggestion
	var/static/list/discount_types = list(
		"small_discount" = 4,
		"average_discount" = 2,
		"big_discount" = 1,
	))

	switch(pick_weight(discount_types))
```

We're going to run this a couple of times for everyone with an uplink so might as well not recreate the same list every time.